x/orderbook/bucket_test.go

```diff
@@ -1,138 +1,289 @@
 				t.Fatalf("unexpected error: %+v", err)
 			}
+			assert.Equal(t, tc.expected, index)
 		})
 	}
 }
 
 func TestOpenOrderIndexer(t *testing.T) {
+	now := weave.AsUnixTime(time.Now())
+	onceUponATime := weave.AsUnixTime(time.Time{})
+
+	orderBookID := weavetest.SequenceID(5)
+	side := Side_Ask
+
+	openOrder := &Order{
+		Metadata:       &weave.Metadata{Schema: 1},
+		Trader:         weavetest.NewCondition().Address(),
+		OrderBookID:    orderBookID,
+		Side:           side,
+		OrderState:     OrderState_Open,
+		OriginalOffer:  coin.NewCoinp(100, 0, "ETH"),
+		RemainingOffer: coin.NewCoinp(50, 17, "ETH"),
+		Price:          NewAmountp(121, 0),
+		CreatedAt:      onceUponATime,
+		UpdatedAt:      now,
+	}
+
+	doneOrder := &Order{
+		Metadata:       &weave.Metadata{Schema: 1},
+		Trader:         weavetest.NewCondition().Address(),
+		OrderBookID:    orderBookID,
+		Side:           side,
+		OrderState:     OrderState_Done,
+		OriginalOffer:  coin.NewCoinp(100, 0, "ETH"),
+		RemainingOffer: coin.NewCoinp(50, 17, "ETH"),
+		Price:          NewAmountp(121, 0),
+		CreatedAt:      onceUponATime,
+		UpdatedAt:      now,
+	}
+
+	cancelledOrder := &Order{
+		Metadata:       &weave.Metadata{Schema: 1},
+		Trader:         weavetest.NewCondition().Address(),
+		OrderBookID:    orderBookID,
+		Side:           side,
+		OrderState:     OrderState_Cancel,
+		OriginalOffer:  coin.NewCoinp(100, 0, "ETH"),
+		RemainingOffer: coin.NewCoinp(50, 17, "ETH"),
+		Price:          NewAmountp(121, 0),
+		CreatedAt:      onceUponATime,
+		UpdatedAt:      now,
+	}
+
+	successCaseExpectedValue := BuildOrderIndexTestCase(openOrder)
+
 	cases := map[string]struct {
-		obj orm.Object
-		wantErr *errors.Error 
+		obj      orm.Object
+		expected []byte
+		wantErr  *errors.Error
 	}{
 		"success": {
-			obj: orm.NewSimpleObj(nil, new(Order)), 
-			wantErr: nil,
+			obj:      orm.NewSimpleObj(nil, openOrder),
+			expected: successCaseExpectedValue,
+			wantErr:  nil,
 		},
-		//TODO BuildOpenOrderIndex cases 
+		"failure, order state done": {
+			obj:      orm.NewSimpleObj(nil, doneOrder),
+			expected: nil,
+			wantErr:  nil,
+		},
+		"failure, order state cancel": {
+			obj:      orm.NewSimpleObj(nil, cancelledOrder),
+			expected: nil,
+			wantErr:  nil,
+		},
+		"failure, obj is nil": {
+			obj:      nil,
+			expected: nil,
+			wantErr:  nil,
+		},
+		// TODO add obj.Value() is nil case
 		"failure not order": {
-			obj: orm.NewSimpleObj(nil, new(OrderBook)),
-			wantErr: errors.ErrState,
+			obj:      orm.NewSimpleObj(nil, new(OrderBook)),
+			expected: nil,
+			wantErr:  errors.ErrState,
 		},
 	}
 
 	for testName, tc := range cases {
 		t.Run(testName, func(t *testing.T) {
-			if _, err := openOrderIndexer(tc.obj); !tc.wantErr.Is(err) {
+			index, err := openOrderIndexer(tc.obj)
+
+			if !tc.wantErr.Is(err) {
 				t.Fatalf("unexpected error: %+v", err)
 			}
+			assert.Equal(t, tc.expected, index)
 		})
 	}
 }
 
+func BuildOrderIndexTestCase(order *Order) []byte {
```

Comment:
This is basically a copy of the code we want to copy...
I would remove this function an do it manually to check.

Reply:
> orderBookID := weavetest.SequenceID(5)
> side := Side_Ask
> Price:          NewAmountp(121, 0)

expected := []byte{0, 0, 0, 0, 0, 0, 0, 5, 1, 0, 0, 0, 0, 0, 0, 0, 121, 0, 0, 0, 0, 0, 0, 0, 0}

Reply:
On it :+1